src/components/Menu/index.js

```diff
@@ -0,0 +1,632 @@
+    type: {
+      type: String,
+      default: 'ghost'
+    },
+
+    // only apply when uses the hasIconOnly property
+    hasIconOnly: Boolean,
+    isRounded: Boolean,
+
+    // only apply when does not use the hasIconOnly property
+    label: {
+      type: String,
+      default: null
+    }
+  },
+
+  computed: {
+    context () {
+      return this.$MenuContext()
+    },
+    isOpen () {
+      return this.context.isOpen || false
+    }
+  },
+
+  methods: {
+    /**
+     * triggers a click event and toggle menu state
+     * @param {Event} event
+     */
+    handleClick (event) {
+      const { closeMenu, focusOnFirstItem } = this.context
+
+      this.$emit('click', event)
+
+      event.preventDefault()
+      event.stopPropagation()
+
+      if (this.isOpen) {
+        closeMenu()
+      } else {
+        focusOnFirstItem()
+      }
+    },
+
+    /**
+     * triggers a keydown event
+     * @param {Event} event
+     */
+    handleKeyDown (event) {
+      const { focusOnFirstItem, focusOnLastItem } = this.context
+
+      this.$emit('keydown', event)
+
+      if (event.key === 'ArrowUp') {
+        event.preventDefault()
+        focusOnLastItem()
+      }
+
+      if (event.key === 'ArrowDown') {
+        event.preventDefault()
+        focusOnFirstItem()
+      }
+    }
+  },
+
+  render (h) {
+    const { isOpen, menuButtonId } = this.context
+
+    if (this.hasIconOnly) {
+      return h(SbButton, {
+        attrs: {
+          ...this.$attrs,
+          id: menuButtonId,
+          'aria-controls': this.menuListId,
+          'aria-haspopup': 'true',
+          'aria-expanded': isOpen ? 'true' : null
+        },
+        props: {
+          isRounded: this.isRounded,
+          hasIconOnly: true,
+          icon: 'overflow-menu-vertic',
+          type: this.type
+        },
+        on: {
+          ...this.$listeners,
+          click: this.handleClick,
+          keydown: this.handleKeyDown
+        }
+      })
+    }
+
+    return h(SbButton, {
+      attrs: {
+        ...this.$attrs,
+        id: menuButtonId,
+        'aria-haspopup': 'true',
+        'aria-expanded': isOpen ? 'true' : null
+      },
+      props: {
+        iconRight: 'chevron-down',
+        label: this.label,
+        type: this.type
+      },
+      on: {
+        ...this.$listeners,
+        click: this.handleClick,
+        keydown: this.handleKeyDown
+      }
+    }, this.$slots.default)
+  }
+}
+
+/**
+ * @vue/component
+ *
+ * SbMenu is the Menu component that triggers with a button
+ */
+const SbMenu = {
+  name: 'SbMenu',
+
+  provide () {
+    return {
+      $MenuContext: () => this.MenuContext
+    }
+  },
+
+  props: {
+    value: {
+      type: Boolean,
+      default: false
+    }
+  },
+
+  data: () => ({
+    activeIndex: -1,
+    focusableElements: [],
+    isOpen: false,
+    menuListId: `sb-menu-list-${randomString(4)}`,
+    menuButtonId: `sb-menu-button-${randomString(4)}`
+  }),
+
+  computed: {
+    MenuContext () {
```

Comment:
Just a matter of standardization: here the variable is starting with a capital letter. Is this correct?

Reply:
Yeah, you're right. Updated!